osu.Game/Skinning/Components/BeatmapInfoDrawable.cs

```diff
@@ -0,0 +1,314 @@
+
+        [SettingSource("Show Value Suffix", "Should the Value Suffix be included?")]
+        public BindableBool ShowValueSuffix { get; } = new BindableBool();
+
+        [Resolved]
+        private IBindable<WorkingBeatmap> beatmap { get; set; } = null!;
+
+        private readonly OsuSpriteText text;
+
+        public LocalisableString TooltipText { get; set; }
+        private LocalisableString value;
+        private LocalisableString labelText;
+
+        public BeatmapInfoDrawable()
+        {
+            InternalChildren = new Drawable[]
+            {
+                text = new OsuSpriteText
+                {
+                    Text = "BeatInfoDrawable",
+                    Anchor = Anchor.CentreLeft,
+                    Origin = Anchor.CentreLeft,
+                    Font = OsuFont.Default.With(size: 40)
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Type.BindValueChanged(update, true);
+            ShowLabel.BindValueChanged(ignored => updateLabel());
+            ValueBeforeLabel.BindValueChanged(ignored => updateLabel());
+            LabelPrefix.BindValueChanged(ignored => updateLabel());
+            ShowLabelPrefix.BindValueChanged(ignored => updateLabel());
+            LabelSuffix.BindValueChanged(ignored => updateLabel());
+            ShowLabelSuffix.BindValueChanged(ignored => updateLabel());
+            ValuePrefix.BindValueChanged(ignored => updateLabel());
+            ShowValuePrefix.BindValueChanged(ignored => updateLabel());
+            ValueSuffix.BindValueChanged(ignored => updateLabel());
+            ShowValueSuffix.BindValueChanged(ignored => updateLabel());
+        }
+
+        private LocalisableString getLabelText()
+        {
+            if (!ShowLabel.Value) return new LocalisableString("");
+
+            return LocalisableString.Format("{0}{1}{2}",
+                ShowLabelPrefix.Value ? LabelPrefix.Value : "",
+                labelText,
+                ShowLabelSuffix.Value ? LabelSuffix.Value : "");
+        }
+
+        private LocalisableString getValueText()
+        {
+            return LocalisableString.Format("{0}{1}{2}",
+                ShowValuePrefix.Value ? ValuePrefix.Value : "",
+                value,
+                ShowValueSuffix.Value ? ValueSuffix.Value : "");
+        }
+
+        private void updateLabel()
+        {
+            text.Text = LocalisableString.Format(
+                ValueBeforeLabel.Value ? "{1}{0}" : "{0}{1}",
+                getLabelText(),
+                getValueText()
+            );
+            Width = text.Width;
+            Height = text.Height;
+        }
+
+        private void update(ValueChangedEvent<BeatmapInfo> type)
+        {
+            switch (type.NewValue)
+            {
+                case BeatmapInfo.CircleSize:
+                    beatmap.BindValueChanged(bm =>
```

Comment:
I believe you're leaking events here, since every time the `Type` is changed you're adding a new `ValueChanged`. A single line fix would be to call `beatmap.UnbindAllEvents()` at the start of this method. A more correct fix would probably be to switch the nesting here so the `switch` is done inside a single `beatmap.BindValueChanged()` which is bound in `LoadComplete`. Something like this:

```csharp
LoadComplete()
{
    type.BindValueChanged(_ => updateBeatmapContent());
    beatmap.BindValueChanged(_ => updateBeatmapContent(), true);
}

private updateBeatmapContent()
{
    switch (type.Value)
    {
        ...
    }
}
```